app/api/auth/views.py

```diff
@@ -0,0 +1,78 @@
+import typing as tp
+
+from drf_yasg.utils import swagger_auto_schema
+
+from django.contrib.auth import get_user_model
+
+from rest_framework import status
+from rest_framework import mixins
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.viewsets import GenericViewSet
+from rest_framework_simplejwt.views import TokenObtainPairView
+
+from api.auth.permissions import IsOwnerOrReadOnlyAccountPermission
+from api.auth.serializers import (
+    RegisterUserSerializer,
+    CustomTokenObtainPairSerializer,
+    AccountModelSerializer,
+    UpdateAccountSerializer,
+)
+from api.auth.services import create_new_user, update_account_data
+
+
+class RegisterUserApiView(APIView):
+    """
+    Api View for creating new user
+    """
+
+    @swagger_auto_schema(request_body=RegisterUserSerializer)
+    def post(self, request: Request, *args: tp.Any, **kwargs: tp.Any) -> Response:
+        serializer = RegisterUserSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer_data = serializer.data
```

Comment:
I would either create a new layer around serializers and call it from here or included in into services

Reply:
like the layer would return serialized and validated data and services layer would take it as input

Reply:
done